src/core/server/pulse/collectors/notifications.ts

```diff
@@ -17,11 +17,83 @@
  * under the License.
  */
 
-import { PulseCollector } from '../types';
+import { PulseCollector, CollectorSetupContext } from '../types';
+
+export interface Payload {
+  records: NotificationInstruction[];
+  deploymentId: string;
+}
+
+export interface NotificationInstruction {
+  hash: string;
+  title: string;
+  status: 'new' | 'seen';
+  description: string;
+  linkUrl: string;
+  linkText: string;
+  badge: string;
+  publishOn: string;
+  expireOn: string;
+  seenOn: string;
+}
+
+function flatMap<T, X>(arr: T[], foldMethod: (x: T) => X | X[]): X[] {
+  return arr.reduce((acc, x) => acc.concat(foldMethod(x)), [] as X[]);
+}
 
 export class Collector extends PulseCollector {
-  public async putRecord() {}
-  public async getRecords() {
+  private readonly channelName = 'notifications';
+  private readonly indexName = '.kibana_pulse_local_notifications';
```

Comment:
Maybe we can publicly expose a static method `buildIndex` so we don't need to maintain the full index name on both ends (not for this PR, but a comment to eventually tackle) :)